### ISSUE TITLE
Suppress nullable warnings for nullable value types with `!`

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#31501]
```C#
class Program
{
    static T F<T>(T? x) where T : struct
    {
        return (T)x!;
    }
}
```
Expected Result: No warnings

Actual Result:
```
(5,19): warning CS8624: The suppression operator (!) can only be applied to reference types.
(5,16): warning CS8629: Nullable value type may be null.
```

**Comment by cston** [comment]
Fixed in #31499.
